### ISSUE TITLE
match: Could not create another certificate, reached the maximum number of available certificates.

### ISSUE THREAD

**Issue by lakamsani** [fastlane/fastlane#5765]
### New Issue Checklist

- [X ] Updated fastlane to the latest version
- [x ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
After running "fastlane nuke development" a couple of times, match gives the error:
Could not create another certificate, reached the maximum number of available certificates.

This is NOT with an enterprise account nor is this in a CI machine. It is on a development laptop. Is there a workaround to manually clear something in the keychain to get past this problem?

##### Complete output when running fastlane, including the stack trace and command used

```
% match development -u <user>   -a <app-id>

+-----------------------+-------------------------------------------+
|                      Summary for match 0.6.2                      |
+-----------------------+-------------------------------------------+
| username              | email@me.com                         |
| app_identifier        | <appid>       |
| git_url               | git@bitbucket.org:mine/certificates.git |
| type                  | development                               |
| git_branch            | master                                    |
| keychain_name         | login.keychain                            |
| readonly              | false                                     |
| team_id               | XXXXXX                                |
| verbose               | false                                     |
| force                 | false                                     |
| shallow_clone         | false                                     |
| force_for_new_devices | false                                     |
| skip_docs             | false                                     |
+-----------------------+-------------------------------------------+

[00:18:03]: Cloning remote git repo...
[00:18:23]: 🔓  Successfully decrypted certificates repo
[00:18:23]: Verifying that the certificate and profile are still valid on the Dev Portal...
[00:18:25]: Couldn't find a valid code signing identity in the git repo for development... creating one for you now

+---------------+------------------------------------------------+
|                     Summary for cert 1.4.1                     |
+---------------+------------------------------------------------+
| development   | true                                           |
| force         | true                                           |
| username      | email@me.com                              |
| team_id       | XXXX                                     |
| keychain_path | /Users/me/Library/Keychains/login.keychain |
+---------------+------------------------------------------------+

[00:18:25]: Starting login with user 'email@me.com '
[00:18:27]: Successfully logged in
[00:18:27]: 🔒  Successfully encrypted certificates repo

Looking for related GitHub issues on fastlane/fastlane...

➡️  Could not create another certificate, reached the maximum number of available certificates.
   https://github.com/fastlane/fastlane/issues/3472 [closed] 9 💬
   08 Mar 2016

➡️  Could not create another certificate, reached the maximum number of available certificates
   https://github.com/fastlane/fastlane/issues/3123 [closed] 10 💬
   08 Mar 2016

➡️  Error Certificate for "AppStore" on Sigh step
   https://github.com/fastlane/fastlane/issues/3900 [closed] 29 💬
   2 weeks ago

and 8 more at: https://github.com/fastlane/fastlane/search?q=Could%20not%20create%20another%20certificate,%20reached%20the%20maximum%20number%20of%20available%20certificates.&type=Issues&utf8=✓

[!] Could not create another certificate, reached the maximum number of available certificates.
```

### Configuration Files

just match command

### Environment

fastlane version (run `fastlane -v`): fastlane 1.100.0, match 0.6.2

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? no

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? no

**Comment by dabvid** [comment]
I got the same problem as you on my laptop but with distribution, on a non enterprise account.

**Comment by lakamsani** [comment]
The workaround for this is to go to the Apple dev portal and delete the certificate associated with the username supplied with the -u option. This is because "fastlane nuke development" does not delete that cert from either the dev portal or from the local keychain. It only seems to drop it from the git repo that match uses (?).  Here are a couple of screen shots showing where I manually deleted the certs to get going. The dev portal one is the one that fixed it. Deleting the one in the keychain is just for keeping things clean. 

<img width="135" alt="screen shot 2016-08-16 at 10 34 21 am" src="https://cloud.githubusercontent.com/assets/3322496/17710318/8d2dae86-63a1-11e6-9cde-68baa35072cc.png">
<img width="243" alt="screen shot 2016-08-16 at 10 33 55 am" src="https://cloud.githubusercontent.com/assets/3322496/17710319/8d2f45b6-63a1-11e6-8738-840301e46833.png">

given that match creates profiles, certs etc in dev portal if they don't exist nuke should probably delete both profiles and certs to avoid this type of (potentially confusing error message?)

**Comment by lakamsani** [comment]
After using fastlane for a while I see a large number unknown private keys entries like this in the keychain. This is after I started using fastlane and experimented  my development laptop during the past 3-4 months. After I 'nuke'd development they were still there so I manually dropped.  Given that match stores keys in the git repo I wonder what are these entries in the keychain.

<img width="999" alt="screen shot 2016-08-17 at 12 47 21 pm" src="https://cloud.githubusercontent.com/assets/3322496/17750760/f3138108-6478-11e6-9706-2cca15ee62cb.png">

Given that I found a workaround for my primary issue, I 'm closing this.
